base/src/com/google/idea/blaze/base/actions/BuildFileUtils.java

```diff
@@ -74,8 +77,16 @@ static PsiElement findBuildTarget(Project project, BlazePackage parentPackage, F
     if (packagePath == null) {
       return null;
     }
+    final Stream<Label> targetLabels = SourceToTargetMap.getInstance(project)
+        .getTargetsToBuildForSourceFile(file).stream();
+
+    LanguageClass fileLanguage = LanguageClass.fromExtension(FileUtilRt.getExtension(file.getName()).toLowerCase());
+    final Stream<Label> maybeFilteredLabels = (fileLanguage == LanguageClass.C) ?
```

Comment:
It would be a great improvement if the name (or more names) communicate what's going on here:
1. filter is applicable only for C
2. what that filter does for that language

Reply:
I thought about extracting a private method to express intent but that is not the style of the code. OTOH if we're not going to upstream it then maybe it's not such a big deal.

Regarding what it does for that language- honestly I'm not sure it's easy for me to explain this in a comment let alone in a short name. Maybe a private method with a long comment with roughly the above quoted info?

Reply:
If you plan to contribute to upstream, then most likely they will inline your methods. I would prefer using at least better name for vars. If this is not enough, please comment.

Reply:
please review now

Reply:
What triggered my attention in the first place was name `maybeFilteredLabels`. Maybe it makes sense to inline it now?

Reply:
I inlined it. Not sure I like it that much because it makes me see even more how much I miss partial application :)
In any case please review

Reply:
Do you still need `.filter(l -> l.blazePackage().equals(packagePath))` after the call to `filterCRelatedLabelsWhichBelongToOtherPackages`?

Reply:
Of course not, I’m an ass :) thanks!
Removed

Reply:
refactoring without tests is not a good idea :)

Reply:
:) we're good?